src/coreclr/vm/ceeload.cpp

```diff
@@ -4484,7 +4480,46 @@ void SaveManagedCommandLine(LPCWSTR pwzAssemblyPath, int argc, LPCWSTR *argv)
         Append_Next_Item(&pCursor, &remainingLen, argv[i], moreArgs);
     }
 
-    s_pCommandLine = pNewCommandLine;
+    if (InterlockedCompareExchangeT<LPCWSTR>(&s_pCommandLine, pNewCommandLine, NULL) != NULL)
+    {
+        delete[] pNewCommandLine;
+    }
+}
+#endif
+
+LPCWSTR GetCommandLineForDiagnostics()
+{
+    // Get the saved managed command line.
+    if (s_pCommandLine != NULL)
+        return s_pCommandLine;
+
+#ifndef TARGET_UNIX
+    // On Windows, use the OS command line which contains the executable and all arguments.
+    s_pCommandLine = GetCommandLineW();
+    return s_pCommandLine;
+#else
+    // Get the command line from the host
+    SString cmdLine;
+    if (HostInformation::GetProperty(HOST_PROPERTY_COMMAND_LINE, cmdLine))
+    {
+        LPCWSTR cmdLineLocal = cmdLine.GetCopyOfUnicodeString();
+        if (InterlockedCompareExchangeT(&s_pCommandLine, cmdLineLocal, NULL) != NULL)
+        {
+            delete cmdLineLocal;
+        }
+
+        return s_pCommandLine;
+    }
+
+    // Get a computed command line if we have saved command line info
+    if (s_assemblyPath != NULL)
+    {
+        CreateCommandLineForDiagnostics(s_assemblyPath, s_appArgc, s_appArgv);
+        return s_pCommandLine;
+    }
+
+    // Fall back to the result from GetCommandLineW() instead
+    return GetCommandLineW();
```

Comment:
It will return the current executable path (same fallback behaviour that `GetCommandLineForDiagnostics` had before this change). I could switch it to `WszGetModuleFileName` - still using Win32 PAL, but representative of what it returns and a more widely used API.

Reply:
I think this depends on what we want this logging to do.

On Windows, this logged the raw process command line, without any post-processing or customization. Is this the behavior we want?

If it is the behavior we like, we can implement it on Unix too - read the command line from `/proc/self/cmdline` on Linux, etc. And we do not need the host customization hook at all.

Reply:
> On Windows, this logged the raw process command line, without any post-processing or customization. Is this the behavior we want?

@davmason do you have a good sense of this? I was going with matching the current non-suspension behaviour (which returns the command line as processed for the managed app)

cc @jander-msft who posted the original issue

Reply:
I think raw command line should be fine, unless @jander-msft says otherwise. The type of tool that wants to read the command line probably isn't picky about formatting